awscli/clidriver.py

```diff
@@ -103,12 +91,28 @@ def main():
     return rc
 
 
+def construct_cli_error_handlers_chain():
+    handlers = [
+        ParamValidationErrorsHandler(),
+        UnknownArgumentErrorHandler(),
+        ClientErrorHandler(),
```

Comment:
Going through the original try/except branching logic, it seems like `ClientError` is handled later. I think it may make sense to lower this to after `InterruptExceptionHandler` to limit the blast radius if the order does actually matter and keep parity.